experimental/segmenter/src/complex.rs

```diff
@@ -83,12 +84,14 @@ pub fn complex_language_segment_utf16(
         }
 
         if let Some(payload) = dictionary.best(str_per_lang[0] as u32) {
-            if let Ok(segmenter) = DictionarySegmenter::try_new_unstable(payload) {
-                let breaks = segmenter.segment_utf16(&str_per_lang);
-                let mut r: Vec<usize> = breaks.map(|n| offset + n).collect();
-                result.append(&mut r);
-                offset += str_per_lang.len();
-                continue;
+            if let Some(grapheme) = grapheme {
+                if let Ok(segmenter) = DictionarySegmenter::try_new_unstable(payload, grapheme) {
+                    let breaks = segmenter.segment_utf16(&str_per_lang);
+                    let mut r: Vec<usize> = breaks.map(|n| offset + n).collect();
+                    result.append(&mut r);
```

Comment:
`suggestion`: you can do `result.extend(breaks.map(...))` without having to use intermediate `Vec`.

Reply:
Thanks.